vault/logical_system.go

```diff
@@ -692,6 +693,42 @@ func NewSystemBackend(core *Core, config *logical.BackendConfig) (logical.Backen
 				HelpSynopsis:    strings.TrimSpace(sysHelp["audited-headers"][0]),
 				HelpDescription: strings.TrimSpace(sysHelp["audited-headers"][1]),
 			},
+			&framework.Path{
+				Pattern: "plugin-catalog/$",
```

Comment:
I wonder if this should be `sys/plugins/catalog` instead. If we are ever going to allow plugin operations that aren't strictly catalog based (maybe some kind of unloading/reloading, or upgrade paths, or something) we'll then have to introduce new top-level `sys` endpoints. Maybe it's better to simply introduce a collection now even if at the moment we have a single endpoint.